lib/view_component/compiler.rb

```diff
@@ -30,7 +30,12 @@ def compile(raise_errors: false, force: false)
         @component.superclass.compile(raise_errors: raise_errors)
       end
 
-      return if gather_template_errors(raise_errors).any?
+      if template_errors.present?
```

Comment:
Would you be up for adding a comment here briefly summarizing the steps to reproduce this issue? I just want to make sure we document it for those considering changing this class in the future.

Reply:
I added a comment below, but I'd be happy to link to this PR here, although it's really several difference pieces that contribute since we'd need to comment here, and in a few other places like the template errors method.

I don't think we need it long-term though, since we should likely "just" use a mutex around the whole compilation step and avoid race conditions popping up in areas that aren't wrapped in a mutex like we're vulnerable to today.